linkerd/meshtls/rustls/src/creds/verify.rs

```diff
@@ -1,23 +1,33 @@
-use std::convert::TryFrom;
-use std::sync::Arc;
-use std::time::SystemTime;
+use std::{convert::TryFrom, sync::Arc};
 use tokio_rustls::rustls::{
     self,
-    client::{self, ServerCertVerified, ServerCertVerifier},
+    client::{
+        self,
+        danger::{ServerCertVerified, ServerCertVerifier},
+    },
+    crypto::WebPkiSupportedAlgorithms,
+    pki_types::{CertificateDer, ServerName, UnixTime},
     server::ParsedCertificate,
-    Certificate, RootCertStore, ServerName,
+    RootCertStore,
 };
 use tracing::trace;
 
-pub(crate) struct AnySanVerifier(Arc<RootCertStore>);
+#[derive(Debug)]
+pub(crate) struct AnySanVerifier {
+    roots: Arc<RootCertStore>,
+    supported: WebPkiSupportedAlgorithms,
+}
 
 impl AnySanVerifier {
     pub(crate) fn new(roots: impl Into<Arc<RootCertStore>>) -> Self {
-        Self(roots.into())
+        Self {
+            roots: roots.into(),
+            supported: rustls::crypto::ring::default_provider().signature_verification_algorithms,
```

Comment:
This change I'm not entirely confident in. The cert verification methods below now take a list of supported algorithms instead of implicitly using the default, and this was the only reasonable way I was able to get the default list that the `WebPkiServerVerifier` that the implementation we copied was using.

Reply:
Can we instead use the restricted set that we know about?

https://github.com/linkerd/linkerd2-proxy/pull/3419/files#diff-7c54eeeb80146328f2d2a680f933677d0645157007e161590456652105f809adL114